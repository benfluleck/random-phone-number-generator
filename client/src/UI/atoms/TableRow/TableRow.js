import React from 'react';

const TableRow = ({ name, value }) => (
  <tr>
    <td className="is-size-3 has-text-weight-light">{name}</td>
    <td className="is-size-3 has-text-weight-light">{value}</td>
  </tr>
)

export default TableRow;
