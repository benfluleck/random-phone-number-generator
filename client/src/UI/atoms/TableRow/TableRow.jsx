import React from 'react';
import PropTypes from 'prop-types';

const TableRow = ({ name, value }) => (
  <tr>
    <th className="has-text-weight-light">{name}</th>
    <td className="is-size-3">{value}</td>
  </tr>
);

TableRow.propTypes = {
  name: PropTypes.oneOfType([ PropTypes.string, PropTypes.node ]).isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number ]).isRequired,
};


export default TableRow;
