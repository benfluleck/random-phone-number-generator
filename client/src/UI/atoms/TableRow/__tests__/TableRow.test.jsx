import React from 'react';

import { shallow } from 'enzyme';
import TableRow from '../TableRow';

const setup = (props) => {
  const defaultProps = {
    name: 'Testing',
    value: 32
  };
  const wrapper = shallow(
    <TableRow {...props} {...defaultProps} />
  );

  return {
    wrapper,
  };
};


describe('Component - Button', () => {
  test('should render', () => {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
