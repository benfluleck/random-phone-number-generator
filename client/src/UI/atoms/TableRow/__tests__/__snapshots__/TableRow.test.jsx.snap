// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow
    name="Testing"
    value={32}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <th
          className="has-text-weight-light"
        >
          Testing
        </th>,
        <td
          className="is-size-3"
        >
          32
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Testing",
          "className": "has-text-weight-light",
        },
        "ref": null,
        "rendered": "Testing",
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 32,
          "className": "is-size-3",
        },
        "ref": null,
        "rendered": 32,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <th
            className="has-text-weight-light"
          >
            Testing
          </th>,
          <td
            className="is-size-3"
          >
            32
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Testing",
            "className": "has-text-weight-light",
          },
          "ref": null,
          "rendered": "Testing",
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 32,
            "className": "is-size-3",
          },
          "ref": null,
          "rendered": 32,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
